lua/luasnip/extras/postfix.lua

```diff
@@ -65,17 +72,16 @@ local function postfix(context, nodes, opts)
 		user_callback = { user_callback, { "nil", "function" } },
 	})
 
-	local match_pattern
-	if type(context) == "string" then
-		context = {
-			trig = context,
-		}
-		match_pattern = matches.default
-	else
-		match_pattern = context.match_pattern or matches.default
-	end
+	local match_pattern = context.match_pattern or matches.default
+
+	context = node_util.wrap_context(context)
+	context = vim.tbl_extend(
+		"keep",
+		context,
+		get_default_context(context.trig, match_pattern)
+	)
```

Comment:
How about we combine the condition-functions if some are passed by the user?

Reply:
How does it look now? The default condition will only run if user doesn't provide a function or if the user provided function returns `nil`.

Reply:
Mhmm almost, I was thinking about something like
```lua
if user_condition then
	context.condition = function(...)
		return user_condition(...) and enable_postfix_condition(...)
	end
else
	context.condition = enable_postfix_condition
end
```
so both are evaluated (snippet is disabled entirely if there is no text to postfix, and is enabled if there is text to postfix, and the user-condition is true)

Reply:
Updated.